torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -423,6 +423,13 @@ def batched_powerSGD_hook(state: PowerSGDState, bucket) -> torch.futures.Future:
 
     # The input tensor is a flattened 1D tensor.
     input_tensor = bucket.get_tensors()[0]
+
+    # Run vanilla allreduce in the first `start_powerSGD_iter` iterations.
+    if state.iter < state.start_powerSGD_iter:
```

Comment:
Do we have unittests in place for `batched_powerSGD_hook`? If not, can we add them?

Reply:
Batched powerSGD is already tested by `test_powerSGD_ddp_comm_hook_nccl` in test_c10d.py.